src/librustc_span/source_map.rs

```diff
@@ -370,11 +370,11 @@ impl SourceMap {
     pub fn doctest_offset_line(&self, file: &FileName, orig: usize) -> usize {
         match file {
             FileName::DocTest(_, offset) => {
-                return if *offset >= 0 {
-                    orig + *offset as usize
-                } else {
+                if *offset < 0 {
                     orig - (-(*offset)) as usize
-                };
+                } else {
+                    orig + *offset as usize
+                }
```

Comment:
Maybe not in this PR but this looks like it could be cleaned up to be orig + offset.abs() as usize or so. (Maybe even detectable by Clippy?)

Reply:
I believe this would change program behaviour since
`let x = 0 - (-(offset)) as usize;` panics in debug mode (attempt to subtract with overflow) while `offset.abs()` does not.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=015cc425ea1143a576c4d60571f2da62

Reply:
Huh. Interesting. I guess that makes sense. I suspect we might never run this code with overflow checking on...

But good catch!